nvtabular/io.py

```diff
@@ -228,55 +235,68 @@ def __init__(
         # Resolve file system
         self.fs = fs or get_fs_token_paths(str(out_dir))[0]
 
-        # create thread queue and locks
-        self.queue = queue.Queue(num_threads)
-        self.write_locks = [threading.Lock() for _ in range(num_out_files)]
+        # Only use threading if num_threads > 0
+        if self.num_threads:
```

Comment:
Should we be running the multithreaded version with num_threads=1 (which it will right now)?

Reply:
Yes.  I went "back and forth" a bit on this, but ultimately decided it makes sense to allow a single-threaded "multithreaded" case for debugging/testing purposes.

However, happy to avoid multithreding for anything < 2